configure.ac

```diff
@@ -495,8 +495,10 @@ AC_ARG_ENABLE(versioning,
 
 if test "$enable_versioning" = "yes" ; then
    ABIVERSIONFLAGS="-version-info \$(ABIVERSION)"
+   ABI_ABIVERSIONFLAGS="-version-info libmpi_abi_so_verion_m4"
 else
    ABIVERSIONFLAGS="-avoid-version"
+   ABI_ABIVERSIONFLAGS="-avoid-version"
```

Comment:
@dalcinl after we started our ABI compatibility initiative, we made the change that git clones and unofficial/prereleases got an ABI version of 0. The intention was only for official releases to have the current ABI version so that unstable library code would not get accidentally linked to applications built with official releases. The thought was that savvy users could use LD_PRELOAD or add symlinks to override the ABI version if they choose, but the default was to add some safeguards around prereleases.